File: opamenu-api/OpaMenu.Domain/DTOs/CashRegister/CashShiftResponseDto.cs
using OpaMenu.Infrastructure.Shared.Enums.Opamenu;

namespace OpaMenu.Domain.DTOs.CashRegister;

public class CashShiftResponseDto
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string? UserName { get; set; }
    public DateTime OpenedAt { get; set; }
    public DateTime? ClosedAt { get; set; }
    public decimal OpeningBalance { get; set; }
    public decimal? ClosingBalance { get; set; }
    public decimal ExpectedBalance { get; set; }
    public ECashShiftStatus Status { get; set; }
    public List<CashMovementResponseDto> Movements { get; set; } = new();
}

File: opamenu-gestor/lib/features/pos/domain/models/cash_shift_response_dto.dart
import 'package:json_annotation/json_annotation.dart';
import '../enums/cash_shift_status.dart';
import 'cash_movement_response_dto.dart';

part 'cash_shift_response_dto.g.dart';

@JsonSerializable()
class CashShiftResponseDto {
  final String id;
  final String userId;
  final String? userName;
  final DateTime openedAt;
  final DateTime? closedAt;
  final double openingBalance;
  final double? closingBalance;
  final double expectedBalance;
  final CashShiftStatus status;
  final List<CashMovementResponseDto> movements;

  CashShiftResponseDto({
    required this.id,
    required this.userId,
    this.userName,
    required this.openedAt,
    this.closedAt,
    required this.openingBalance,
    this.closingBalance,
    required this.expectedBalance,
    required this.status,
    this.movements = const [],
  });

  factory CashShiftResponseDto.fromJson(Map<String, dynamic> json) =>
      _$CashShiftResponseDtoFromJson(json);

  Map<String, dynamic> toJson() => _$CashShiftResponseDtoToJson(this);
}
